src/services/main/index.ts

```diff
@@ -432,6 +433,22 @@ export default class {
         this.nostrSend({ type: 'app', appId: invoice.linkedApplication.app_id }, { type: 'event', event }, zapInfo.relays || undefined)
     }
 
+    async switchProviderRelay() {
+        const log = getLogger({ component: "main" })
+        log("switching provider relay to backup due to rugpull detection")
+
+        // Swap primary and backup provider relays
+        const currentSettings = this.settings.getSettings().nostrRelaySettings
+        const temp = currentSettings.providerRelay
+        currentSettings.providerRelay = currentSettings.providerRelayBackup
+        currentSettings.providerRelayBackup = temp
```

Comment:
Hello! nice work on the PR, implementation looks very good, but there are a couple issues to address first

Here changing the setting value like this is not a good idea, we do this only in tests, while running you need to create an update function for the value in the settings manager, this way the DB value of the setting updates too, and make it persist after a restart

Tho, I would just get rid of this logic completely, for updating the relay if the provider changes it, we'll use the beacon, the provider before changing relay will tell us via a beacon to what relay it is moving to, with that, we can update the setting